### PR TITLE
fix cdc-bank eventual consistency issue.

### PR DESCRIPTION
**Pull request by JinLingChristopher**
### What problem does this PR solve? 

Since TiCDC is built on the eventual consistency model,  cdc-bank test case may failed due to sync latency, try to fix it.

### Check List 

Tests 

 - No code

Code changes

 - Has Go code change

Side effects

Related changes

### Does this PR introduce a user-facing change?:

NONE

### CODE REVIEW

File: tests/cdc-bank/bank.go [hunk 1]
```diff
@@ -215,7 +215,8 @@ func (c *client) Start(ctx context.Context, cfg interface{}, clientNodes []clust
 						continue
 					}
 					if balanceSum != expectedSum {
-						log.Fatalf("[cdc-bank] [validatorId=%d] sum: %d, expected: %d, startTS: %d", idx, balanceSum, expectedSum, startTS)
+						log.Errorf("[cdc-bank] [validatorId=%d] sum: %d, expected: %d, startTS: %d", idx, balanceSum, expectedSum, startTS)
+						continue
```
Comment (zyguan) [inline tests/cdc-bank/bank.go:219]: Where is the final check?

File: tests/cdc-bank/bank.go [hunk 2]
```diff
@@ -214,15 +216,22 @@ func (c *client) Start(ctx context.Context, cfg interface{}, clientNodes []clust
 						_ = txn.Rollback()
 						continue
 					}
-					if balanceSum != expectedSum {
-						log.Fatalf("[cdc-bank] [validatorId=%d] sum: %d, expected: %d, startTS: %d", idx, balanceSum, expectedSum, startTS)
+					if balanceSum == expectedSum {
+						atomic.AddInt32(&counter, 1)
+						_ = txn.Rollback()
+						log.Infof("[cdc-bank] [validatorId=%d] success, startTS: %d", idx, startTS)
+						return
```
Comment (zyguan) [inline tests/cdc-bank/bank.go:223]: It breaks the original test logical. We want to repeatedly check the total balance during the test (before the `ctx` done), but now it only check `validateConcurrency` times at the beginning.

**Comment by JinLingChristopher** [comment]
The latest following the assumption that "at least one goroutine cannot get expect sum at once".

This assumption is quite weak, for a buggy TiCDC, it may meet the assumption luckily.

It's hard for me to figure out the condition which can guarantee the correctness - - ····

File: tests/cdc-bank/bank.go [hunk 3]
```diff
@@ -192,9 +216,14 @@ func (c *client) Start(ctx context.Context, cfg interface{}, clientNodes []clust
...
 			for {
 				select {
-				case <-ctx.Done():
-					return
-				default:
+				case tblName := <-tblChan:
+					waitCtx, waitCancel := context.WithTimeout(ctx, 2*time.Minute)
+					waitTable(waitCtx, downstream, tblName)
+					waitCancel()
+					log.Infof("ddl synced")
```
Comment (amyangfei) [inline tests/cdc-bank/bank.go:223]: When the validator receives notification from `tblChan`, it seems more DMLs after this DDL could have been executed in upstream?

File: tests/cdc-bank/bank.go [hunk 4]
```diff
@@ -163,23 +184,26 @@ func (c *client) Start(ctx context.Context, cfg interface{}, clientNodes []clust
...
-						log.Errorf("[cdc-bank] [connID=%d] update account2 error %v", connID, err)
-						_ = txn.Rollback()
-						continue
-					}
-					if err := txn.Commit(); err != nil {
-						log.Errorf("[cdc-bank] [connID=%d] commit error %v", connID, err)
-						_ = txn.Rollback()
-						continue
+
+					for atomic.LoadInt32(&skip) != 1 {
```
Comment (amyangfei) [inline tests/cdc-bank/bank.go:188]: If all goroutines enter the for loop, the `skip` will not be set to 1

File: tests/cdc-bank/bank.go [hunk 5]
```diff
@@ -207,22 +241,113 @@ func (c *client) Start(ctx context.Context, cfg interface{}, clientNodes []clust
...
+		case <-ctx.Done():
+			return
+		case <-time.After(time.Second):
+		}
+	}
+}
+
+func isTableExist(ctx context.Context, db *sql.DB, table string) bool {
+	// if table is not exist, return true directly
+	query := fmt.Sprintf("SHOW TABLES LIKE '%s'", table)
```
Comment (amyangfei) [inline tests/cdc-bank/bank.go:309]: `show tables` should `use db` first or use `show tables in db`

**Review by amyangfei — APPROVED** [review]

**Pull request merged**